packages/host/app/lib/formatted-message/utils.ts

```diff
@@ -110,3 +108,44 @@ export function wrapLastTextNodeInStreamingTextSpan(
   }
   return htmlSafe(doc.body.innerHTML);
 }
+
+export interface HtmlTagGroup {
+  type: 'pre_tag' | 'non_pre_tag';
+  content: string;
+}
+
+export function parseHtmlContent(htmlString: string): HtmlTagGroup[] {
+  let result: HtmlTagGroup[] = [];
+
+  let doc = document.createElement('div');
```

Comment:
I think it would be helpful to add a comment explaining why exactly we are creating an element instead of just parsing the string directly (the way it was done previously)